apps/api/src/app/feeds/usecases/delete-feed/delete-feed.usecase.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { FeedRepository, FeedEntity, DalException } from '@novu/dal';
+import { DeleteFeedCommand } from './delete-feed.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class DeleteFeed {
+  constructor(private feedRepository: FeedRepository) {}
+
+  async execute(command: DeleteFeedCommand): Promise<FeedEntity[]> {
+    try {
+      await this.feedRepository.delete({
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _id: command.feedId,
+      });
+    } catch (e) {
+      if (e instanceof DalException) {
+        throw new ApiException(e.message);
+      }
+      throw e;
+    }
```

Comment:
Are we doing a soft DELETE here? What will happen to existing feed items pointing to the deleted Feed?